### PR TITLE
chore(azure): Update tests to support V3 provider

### PR DESCRIPTION
**Pull request by vdmgolub**

### CODE REVIEW

File: internal/providers/terraform/azure/app_service_certificate_order.go [hunk 1]
```diff
@@ -26,6 +26,7 @@ func NewAzureRMAppServiceCertificateOrder(d *schema.ResourceData, u *schema.Usag
 	if d.Get("product_type").Type != gjson.Null {
 		productType = d.Get("product_type").String()
 	}
+	productType = strings.ToLower(productType)
```
Comment (vdmgolub) [inline internal/providers/terraform/azure/app_service_certificate_order.go:29]: The cost components use case-insensitive filter, this is used for breakdown output only.

File: internal/providers/terraform/azure/container_registry.go [hunk 2]
```diff
@@ -40,22 +39,20 @@ func NewAzureRMContainerRegistry(d *schema.ResourceData, u *schema.UsageData) *s
 		includedStorage = decimalPtr(decimal.NewFromFloat(500))
 	}
 
+	locationsCount = len(d.Get("georeplications").Array())
```
Comment (vdmgolub) [inline internal/providers/terraform/azure/container_registry.go:42]: TF expects multiple blocks, previous implementation expected a `location` array instead.

File: internal/providers/terraform/azure/testdata/firewall_test/firewall_test.tf [hunk 3]
```diff
@@ -48,6 +48,8 @@ resource "azurerm_firewall" "standard" {
   name                = "testfirewall"
   location            = "eastus"
   resource_group_name = azurerm_resource_group.example.name
+  sku_name            = "AZFW_Hub"
+  sku_tier            = "Standard"
```
Comment (vdmgolub) [inline internal/providers/terraform/azure/testdata/firewall_test/firewall_test.tf:52]: Required attributes now: https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/guides/3.0-upgrade-guide#resource-azurerm_firewall

NB: TF docs incorrectly mention them as optional: https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/firewall#sku_name

File: internal/providers/terraform/azure/testdata/frontdoor_test/frontdoor_test.tf [hunk 4]
```diff
@@ -11,7 +11,6 @@ resource "azurerm_resource_group" "example" {
 resource "azurerm_frontdoor" "frontdoor_without_usage" {
   name                                         = "exampleFrontdoor"
   resource_group_name                          = azurerm_resource_group.example.name
-  enforce_backend_pools_certificate_name_check = false
```
Comment (vdmgolub) [inline internal/providers/terraform/azure/testdata/frontdoor_test/frontdoor_test.tf:14]: Deprecated and removed: https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/guides/3.0-upgrade-guide#resource-azurerm_frontdoor.

Not used in the resource, decided to just remove.

File: internal/providers/terraform/azure/testdata/hdinsight_hadoop_cluster_test/hdinsight_hadoop_cluster_test.tf [hunk 5]
```diff
@@ -35,7 +35,6 @@ resource "azurerm_hdinsight_hadoop_cluster" "with_edge" {
   }
 
   gateway {
-    enabled  = true
```
Comment (vdmgolub) [inline internal/providers/terraform/azure/testdata/hdinsight_hadoop_cluster_test/hdinsight_hadoop_cluster_test.tf:38]: This and other changes: deprecated and removed https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/guides/3.0-upgrade-guide#resource-azurerm_hdinsight_hadoop_cluster

File: internal/providers/terraform/azure/testdata/kubernetes_cluster_test/kubernetes_cluster_test.tf [hunk 6]
```diff
@@ -64,19 +64,16 @@ resource "azurerm_kubernetes_cluster" "paid_5nc_32gb" {
...
+  resource_group_name              = azurerm_resource_group.example.name
+  dns_prefix                       = "exampleaks1"
+  sku_tier                         = "Paid"
+  http_application_routing_enabled = true
+
   network_profile {
     network_plugin    = "azure"
-    load_balancer_sku = "Standard"
-  }
-  addon_profile {
```
Comment (vdmgolub) [inline internal/providers/terraform/azure/testdata/kubernetes_cluster_test/kubernetes_cluster_test.tf:76]: Deprecated and remove, `http_application_routing_enabled` is already used in 2.x: https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/guides/3.0-upgrade-guide#resource-azurerm_kubernetes_cluster

File: internal/providers/terraform/azure/testdata/lb_outbound_rule_test/lb_outbound_rule_test.tf [hunk 7]
```diff
@@ -26,13 +26,11 @@ resource "azurerm_lb" "example" {
 }
 
 resource "azurerm_lb_backend_address_pool" "example" {
-  resource_group_name = azurerm_resource_group.example.name
   loadbalancer_id     = azurerm_lb.example.id
   name                = "be-%d"
 }
 
 resource "azurerm_lb_outbound_rule" "rules" {
-  resource_group_name     = azurerm_resource_group.example.name
```
Comment (vdmgolub) [inline internal/providers/terraform/azure/testdata/lb_outbound_rule_test/lb_outbound_rule_test.tf:35]: Deprecated and removed, added v2 tests. https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/guides/3.0-upgrade-guide#resource-azurerm_lb_outbound_rule

File: internal/providers/terraform/azure/testdata/lb_rule_test/lb_rule_test.tf [hunk 8]
```diff
@@ -25,7 +25,6 @@ resource "azurerm_lb" "example" {
 }
 
 resource "azurerm_lb_rule" "rules" {
-  resource_group_name            = azurerm_resource_group.example.name
```
Comment (vdmgolub) [inline internal/providers/terraform/azure/testdata/lb_rule_test/lb_rule_test.tf:28]: Deprecated and removed https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/guides/3.0-upgrade-guide#resource-azurerm_lb_rule

File: internal/providers/terraform/azure/testdata/log_analytics_workspace_test/log_analytics_workspace_test.tf [hunk 9]
```diff
@@ -30,15 +30,6 @@ resource "azurerm_log_analytics_workspace" "capacity_gb_data_ingestion" {
   reservation_capacity_in_gb_per_day = 100
 }
 
-resource "azurerm_log_analytics_workspace" "capacity_typo_gb_data_ingestion" {
-  name                = "acctest-03"
-  location            = azurerm_resource_group.example.location
-  resource_group_name = azurerm_resource_group.example.name
-  sku                 = "CapacityReservation"
-  # this typo is deliberate see: https://github.com/hashicorp/terraform-provider-azurerm/pull/14910 for more info
-  reservation_capcity_in_gb_per_day = 100
```
Comment (vdmgolub) [inline internal/providers/terraform/azure/testdata/log_analytics_workspace_test/log_analytics_workspace_test.tf:39]: Deprecated and removed. Kept in the code for now.

File: internal/providers/terraform/azure/testdata/sql_managed_instance_test/sql_managed_instance_test.tf [hunk 10]
```diff
@@ -19,7 +19,7 @@ resource "azurerm_subnet" "example" {
   name                 = "subnet-mi"
   resource_group_name  = azurerm_resource_group.example.name
   virtual_network_name = azurerm_virtual_network.example.name
-  address_prefix       = "10.0.0.0/24"
+  address_prefixes     = ["10.0.0.0/24"]
```
Comment (vdmgolub) [inline internal/providers/terraform/azure/testdata/sql_managed_instance_test/sql_managed_instance_test.tf:22]: https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/guides/3.0-upgrade-guide#resource-azurerm_subnet

File: internal/providers/terraform/azure/testdata/synapse_spark_pool_test/synapse_spark_pool_test.tf [hunk 11]
```diff
@@ -39,6 +39,10 @@ resource "azurerm_synapse_workspace" "example" {
     tenant_id = "00000000-0000-0000-0000-000000000000"
   }
 
+  identity {
+    type = "SystemAssigned"
+  }
+
```
Comment (vdmgolub) [inline internal/providers/terraform/azure/testdata/synapse_spark_pool_test/synapse_spark_pool_test.tf:45]: Required, related mention: https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/guides/3.0-upgrade-guide#resource-azurerm_synapse_workspace_key

**Review by aliscott — APPROVED** [review]
LGTM!

**Pull request merged**